### PR TITLE
Awful poll function

### PR DESCRIPTION
**Pull request by h-anjru**
Don't laugh

### CODE REVIEW

File: crimsobot/cogs/utilities.py [hunk 1]
```diff
@@ -14,6 +16,111 @@ class Utilities(commands.Cog):
...
+
+        poll_body = ['**{}**\n'.format(question.strip())]
+        for idx, choice in enumerate(choices):
+            poll_body.append('{} {}'.format(poll_pool[idx], choice.strip()))
+
+        embed = c.crimbed(
+            title='{} has created a poll!'.format(ctx.author),
+            descr='{}'.format('\n'.join(poll_body)),
+            thumb_name='think',
+            footer='\u200dPoll ID: {d.month}{d.day}{d.hour}{d.minute}{d.second}'.format(d=datetime.utcnow())
```
Comment (williammck) [inline crimsobot/cogs/utilities.py:46]: only real issue i see here is conflicts between polls created in the same second

File: crimsobot/cogs/utilities.py [hunk 2]
```diff
@@ -14,6 +18,113 @@ class Utilities(commands.Cog):
...
+        )
+
+        msg = await ctx.send(embed=embed)
+
+        # add reactions to msg
+        for emoji in poll_pool[0:len(choices)]:
```
Comment (kaylynn234) [inline crimsobot/cogs/utilities.py:52]: `0:len(choices)` can just be `:len(choices)`

File: crimsobot/cogs/utilities.py [hunk 3]
```diff
@@ -14,6 +16,111 @@ class Utilities(commands.Cog):
...
+        You can only tally the results of a poll in the same channel as the poll.
+        Tallying a poll does not close the poll; its results can be tallied again if it's not too old."""
+
+        # so this is kinda hacky...
+        # first, check channel's message history (backwards from present) for prior polls from bot
+        try:
+            async for message in ctx.channel.history(limit=10000):
+                if (message.author.id == self.bot.user.id and
+                        len(message.embeds) != 0 and
+                        '\u200dPoll ID:' in message.embeds[0].footer.text):
```
Comment (williammck) [inline crimsobot/cogs/utilities.py:72]: am i too sleepy or do you not need parentheses here

File: crimsobot/cogs/utilities.py [hunk 4]
```diff
@@ -14,6 +18,113 @@ class Utilities(commands.Cog):
...
+    @commands.command(aliases=['tally'], brief='Tally results of a poll!')
+    async def polltally(self, ctx: commands.Context, poll_id: Optional[str] = None) -> None:
+        """Tally the results of the most recent poll in a channel, or of a previous poll if you give the poll's ID.
+        You can only tally the results of a poll in the same channel as the poll.
+        Tallying a poll does not close the poll; its results can be tallied again if it's not too old."""
+
+        # so this is kinda hacky...
+        # first, check channel's message history (backwards from present) for prior polls from bot
+        try:
+            async for message in ctx.channel.history(limit=10000):
```
Comment (kaylynn234) [inline crimsobot/cogs/utilities.py:69]: This is potentially performing 100 requests to the history endpoint per invocation. (100 pages of 100 messages from the API, each "page" is another request.)

Maybe tone this down a bit? I can imagine lots of ratelimits (and potentially cloudflare bans) if the command gets spammed.

File: crimsobot/cogs/utilities.py [hunk 5]
```diff
@@ -14,6 +16,111 @@ class Utilities(commands.Cog):
...
+            await ctx.send('`Poll ID not found, or poll is too old.`')
+            return
+
+        # both the question and the choices will be in here; split into list
+        descr = descr.split('\n')
+
+        # find where the options begin; separate questions and options
+        for idx, option in enumerate(descr):
+            if '1️⃣' in option:
+                options_begin = idx
```
Comment (williammck) [inline crimsobot/cogs/utilities.py:91]: note to self: `options_begin` is line where first reacc is

File: crimsobot/cogs/utilities.py (hunk 4 above)
Comment (williammck) [reply crimsobot/cogs/utilities.py:69]: ugh really?

File: crimsobot/cogs/utilities.py (hunk 3 above)
Comment (h-anjru) [reply crimsobot/cogs/utilities.py:72]: PEP8 recommends () for multiline if statements: ```# Add some extra indentation on the conditional continuation line.
if (this_is_one_thing
        and that_is_another_thing):
    do_something()```

Comment (williammck) [reply crimsobot/cogs/utilities.py:72]: ok so that answers my question of `am i too sleepy`, thanks

File: crimsobot/cogs/utilities.py (hunk 4 above)
Comment (h-anjru) [reply crimsobot/cogs/utilities.py:69]: Well fffff this will have to do for now

File: crimsobot/cogs/utilities.py (hunk 1 above)
Comment (h-anjru) [reply crimsobot/cogs/utilities.py:46]: I uhhhh will take my chances until I come up with something more clever. (I'm gonna eventually cave to using a db I just know it)

**Pull request merged**